### DISCUSSION TITLE
Correct usage of batch_count and parallel_testrun_count

### DISCUSSION THREAD

**Discussion by ssharma0312** [lyndsey-ferguson/fastlane-plugin-test_center#349] [Q&A] [answered]
Should `batch_count` and `parallel_testrun_count` be same? 

Currently I have both `parallel_testrun_count` and `batch_count` set to value `5` with `try_count=3`.  I have over 500 XCTests, At max, I see 3 simulators running tests, 2 usually sitting idle, and I noticed in last few minutes only 1 device is re-running the failed tests. Can I maybe change `batch_count` to `10` split tests into smaller batches so `multi_scan` can take advantage of all 5 simulators started?

```
multi_scan(
      workspace: "XXX.xcworkspace",
      scheme: options[:scheme],
      output_types: 'junit,html',
      output_files: "junit.xml,report.html",
      try_count: 3, # retry _failing_ tests up to three times^1.
      batch_count: 5,
      code_coverage: true,
      parallel_testrun_count: 5, # run subsets of your tests on parallel simulators^2
      testrun_completed_block: test_run_block,
      devices: "iPhone 11",
      result_bundle: "TestResults", # To generate test reports
      output_directory: "./fastlane/test_output/" + options[:branch]
    )
```

**Comment by lyndsey-ferguson** [comment, marked as answer]
Each batch should be running on a simulator, unless some of the simulators have hung and haven't returned yet. 

If you're interested, [here's the code](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb#L246). Do you logs indicate which "workers" are busy and which are not? Splitting into more batches may help, but if you don't provide any "batch" count, the tests will be split into batches: the same number of parallel test runs.
